### PR TITLE
Closes #154

### LINKED ISSUE #154: `get_library()` incorrectly returns non-function objects if found
### What happened?

If an object is set that is the same name as a function, and that object is higher in the search path than the function, logrx records that you have used this object, instead of recording the function you actually used.

Recommend updating so `get_library()` will check the object structure for functions [here](https://github.com/pharmaverse/logrx/blob/main/R/get.R#L220).

### Session Information

N/A

### Reproducible Example

```r
write_file('search <- "dummy object"', "dummy.R", ".")

sys.source("dummy.R", envir = attach(NULL, name = "dummy"))

search()
```

axecute the code above and logrx will show you used dummy:search instead of base:search

``` 
--------------------------------------------------------------------------------
-                          Used Package and Functions                          -
--------------------------------------------------------------------------------
{dummy} search
{package:base} sys.source, attach
```

### PR DESCRIPTION
**Pull request by nicholas-masel**
Closes #154 
### Thank you for your Pull Request! 

We have developed a Pull Request template to aid you and our reviewers.  Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the logrx codebase remains robust and consistent.  

### The spirit of logrx

While many packages to facilitate the logging of code already exist in the R ecosystem, it is hard to find a solution that works well for clinical programming applications. Many logging implementations are more implicit and rely on user input to create the log for the execution of a script. While this is useful for logging specific events of an application, in clinical programming a log has a set purpose.

logrx is built around the concept of creating a log for the execution of an R script that provides an overview of what happened as well as the environment that it happened in. We set out to create a flexible logging utility that could provide the necessary information to anyone reviewing the code execution so they can recreate the execution environment and run the code for themselves. Please make sure your Pull Request meets this **spirit of logrx**.

Please check off each taskbox as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `dev` branch until you have checked off each task.

- [x] The spirit of logrx is met in your Pull Request
- [x] Check that your Pull Request is targeting the dev branch, Pull Requests to master should use the [Release Pull Request Template](https://github.com/pharmaverse/logrx/blob/main/.github/PULL_REQUEST_TEMPLATE/release.md)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests.  Remember to remove any configured log objects at the end of every test using `log_remove()`.
- [x] Creation/updates to relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Address any updates needed for vignettes and/or templates
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-get.R [hunk 1]
```diff
@@ -111,6 +111,25 @@ test_that("used functions returned correctly when file doesn't contain all token
    expect_identical(get_used_functions(filename), expected)
 })
 
+test_that("get_library correctly returns correct function when a non-function
```
Comment (nicholas-masel) [inline tests/testthat/test-get.R:114]:
```suggestion
test_that("get_library returns correct function when a non-function
```

**Comment by nicholas-masel** [comment]
@kodesiba @bms63 I'm not a huge fan of what I had to do to get the examples to run clean, see this [comment ](https://github.com/pharmaverse/logrx/pull/155/files#diff-be75f6398b6213cfe50e1318d8488ba76de0c0cefddbaae032844bad84fe8492R226) for reference.

An alternative would be to add dontrun to the roxygen header for axecute, but I don't like this option either.

**Comment by bms63** [comment]
I'm good with this.  Will this solve another issue that I am having - where if I use data("admiral_vs") and axecute with my code I get these two datasets spilling out into my global environment?  
image

**Comment by nicholas-masel** [comment]
> I'm good with this. Will this solve another issue that I am having - where if I use data("admiral_vs") and axecute with my code I get these two datasets spilling out into my global environment? image

I don't think so.  Looking at `data()`, the `envir` argument defaults to `.GlobalEnv`

```r
data(..., list = character(), package = NULL, lib.loc = NULL,
     verbose = getOption("verbose"), envir = .GlobalEnv,
     overwrite = TRUE)
```

**Pull request merged**